### ISSUE TITLE
Allow users to authenticate over HTTP

### ISSUE THREAD

**Issue by hexacyanide** [sdevgroup/instance-server#1]
This is the server-side part of sdevgroup/game-client#1. The server should be able to accept a `POST` request, and then respond with JSON that indicates success or failure. The client should be written in such a way that connection failures are handled gracefully.

**Comment by nokynokes** [comment]
Should we provides a way to read and write the HTTP cookie headers using the cookie library, like serialization?

**Comment by hexacyanide** [comment]
No, the HTTP headers should be for the server/browser to handle without manual modification. We're using server-side cookies here, so all authorization needs to do is set the appropriate `req.session` properties on successful sign-on so that they can be retrieved when the game client attempts to reuse the authentication credentials over websocket.

**Comment by nokynokes** [comment]
So, for the middle ware, should I be using session.js?

**Comment by hexacyanide** [comment]
You'll want to use [express-session](https://github.com/expressjs/session). Just have a `req.post` route that sets a property, maybe `req.session.token` if a user signs on properly.
